nifi-extension-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/kinesis/stream/ConsumeKinesisStream.java

```diff
@@ -847,4 +852,23 @@ private Date getStartStreamTimestamp(final PropertyContext context, final DateTi
     protected KinesisAsyncClientBuilder createClientBuilder(final ProcessContext context) {
         return KinesisAsyncClient.builder();
     }
+
+    @Override
+    protected void customizeAsyncHttpClientBuilderConfiguration(
+            final ProcessContext context,
+            final NettyNioAsyncHttpClient.Builder builder,
+            final Class<? extends AwsClientBuilder> customizationTargetClass) {
+        if (KinesisAsyncClientBuilder.class.isAssignableFrom(customizationTargetClass)) {
+            // suggested values from KinesisClientUtil
+            final int initialWindowSizeBytes = 512 * 1024; // 512 KB
+            final long healthCheckPingPeriodMillis = 60 * 1000;
```

Comment:
Instead of defining this and then calling `Duration.ofMillis()`, the `Duration` could be declared as a static member variable as `Duration.ofMinutes(1)` named `HEALTH_CHECK_PING_PERIOD`.